app/src/main/java/org/p2p/wallet/auth/ui/smsinput/inputblocked/OnboardingGeneralErrorTimerPresenter.kt

```diff
@@ -0,0 +1,67 @@
+package org.p2p.wallet.auth.ui.smsinput.inputblocked
+
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.ui.smsinput.inputblocked.OnboardingGeneralErrorContract.View.SourceScreen
+import org.p2p.wallet.common.mvp.BasePresenter
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onEach
+
+private const val TIMER_VALUE_FORMAT = "mm:ss"
+private const val START_TIMER_VALUE_MIN = 5
+
+class OnboardingGeneralErrorTimerPresenter :
+    BasePresenter<OnboardingGeneralErrorContract.View>(),
+    OnboardingGeneralErrorContract.Presenter {
+
+    private var sourceScreen = SourceScreen.PHONE_NUMBER_ENTER
+
+    override fun attach(view: OnboardingGeneralErrorContract.View) {
+        super.attach(view)
+
+        createTimerFlow()
+            .onEach(::onTimerValueChanged)
+            .onCompletion { this@OnboardingGeneralErrorTimerPresenter.view?.navigateToStartingScreen() }
+            .launchIn(this)
```

Comment:
Is it okay for us, if the timer will refresh if we re-enter the screen?

Reply:
yes, we can't re-enter this screen before timer runs out

Reply:
yes, we can't re-enter this screen before timer runs out